.circleci/config.yml

```diff
@@ -11,25 +11,53 @@ jobs:
       - checkout
 
       #TODO: add caching once demos share the same requirements file and can live in the same venv
-      - run: &install-and-test-template
-          name: install dependences and run tests
+      #TODO: remove hard code in for-loops
+      - run: &install-dependences-template
+          name: install dependences
           command: |
             shopt -s extglob
 
             # Looping through demos
             # Note: code examples within 'examples' dir share a virtual environment
-            for demo in !(tests|circuit-fault-diagnosis); do
-              if [ -d "$demo" ]; then
+            for demo in !(tests|circuit-fault-diagnosis|qboost); do
+              if [ -d "$demo" ]; then (
                 env="envs/$demo"
                 python -m virtualenv "$env"
                 . "$env/bin/activate"
 
                 pip install -r "$demo/requirements_tests.txt"
+                )
+              fi
+            done
+
+      - run: &run-tests-template
+          name: run unit tests
+          command: |
+            shopt -s extglob
+
+            for demo in !(envs|tests|circuit-fault-diagnosis|qboost|examples); do
+              if [ -d "$demo" ]; then (
+                env="envs/$demo"
+                . "$env/bin/activate"
+
+                cd "$demo"
                 coverage run -m unittest discover
+                )
+              fi
+            done
 
-                deactivate
+            # Loop through examples/*
+            (
```

Comment:
Usually, you would indent under subshell `(..)` - that is a nested scope, afterall:
```
(
    # bla bla
    # more blah
)
```

Reply:
Applies to the above two instances as well.